lib/elixir/lib/task.ex

```diff
@@ -540,6 +540,23 @@ defmodule Task do
   If for some reason you cannot take the elements before hand,
   you can use `:max_concurrency` to limit how many elements
   may be over processed at the cost of reducing concurrency.
+
+  ## First async task to complete
+
+  Given a list of async functions, you can get the result of the first async task to complete.
+
+      [
+        &heavy_call_1/0,
+        &heavy_call_2/0,
+        &heavy_call_3/0
+      ]
+      |> Task.async_stream(fn fun -> fun.() end, ordered: false)
+      |> Stream.filter(&match?({:ok, _}, &1))
+      |> Stream.take(1)
+      |> Enum.to_list()
```

Comment:
Maybe I'm missing something, but isn't this equivalent to

```
|> Task.async_stream(fn fun -> fun.() end, ordered: false)
|> Stream.filter(&match?({:ok, _}, &1))
|> Enum.take(1)
```

Or even better an `|> Enum.at(0)` if you want `value | nil`?

Reply:
@benwilson512 The `Stream.take(1)` as otherwise we may cause more items than requested to be processed. https://hexdocs.pm/elixir/Task.html#async_stream/5-attention-async-take

However `Enum.at(0)` in the place of `Enum.to_list()` would make more sense as we want to get the value directly.

Reply:
I think the point in the docs is placing the `take` call before the `async_stream` call, less about the `Enum` vs `Stream` module - to limit the number of tasks that are launched, rather then the number of results collected.

Reply:
Yeah, you both are completely right, I misread the example.